src/PluginBase.php

```diff
@@ -64,5 +103,39 @@ public function createInstaller(string $pluginName, string $keyDirectory, Packag
 		);
 	}
 
-	abstract public function installOrUpdateFunction(PackageEvent $event) : void;
+	/**
+	 * Returns the config settings
+	 *
+	 * Tries to load configuration from a mediator.json file. Settings defined
+	 * in code overwrite settings defined in the configuration file.
+	 *
+	 * @throws \JsonException
+	 */
+	private function getConfig(): array
+	{
+		$config     = [];
+		$fileValues = is_file($this->configPath)
+		            ? $this->loadConfig()
+		            : self::$defaultConfig;
```

Comment:
Is it a clever idea to expect a variable here? What happens if the variable is not set?
Wouldn't an `abstract protected function getDefaultConfig(): Configuration` a better idea to make sure that a path is set?